lib/folio/holdings.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+module Folio
+  class Holdings
+    YEAR_REGEX = /(?:19|20)\d{2}/
+    # We've seen chronologies that looks like full dates, MON YYYY, WIN YYYY, and nil
+    def self.find_latest(holdings)
+      # NOTE: We saw some piece records without 'chronology'. Was this just test data?
+      pieces = holdings.filter_map do |piece|
+        piece.merge(sortable_date: extract_sortable_date(piece['chronology']) || extract_sortable_date(piece['enumeration']),
+                    sortable_tokens: extract_sortable_tokens(piece['chronology']) || extract_sortable_tokens(piece['enumeration']))
+      end
+      pieces.max_by { |piece| [piece.fetch(:sortable_date), piece.fetch(:sortable_tokens)] }
```

Comment:
In #905, I suggested step d to fall back on a text sort. Was that problematic, and if so, could you drop a comment on #905 with why?

Reply:
I've added a fallback to text sort now.